webapp/components/rhs_comment.jsx

```diff
@@ -327,12 +338,37 @@ export default class RhsComment extends React.Component {
             );
     }
 
+    emojiPickerClick() {
+        let reactOffset = 15;
+        const reactHeight = 360;
+        const reactionIconY = ReactDOM.findDOMNode(this).getBoundingClientRect().top;
+        const rhsHeight = 700; // ReactDOM.findDOMNode(this).parent.parent.parent.getBoundingClientRect().height;
+        const spaceAvail = rhsHeight - reactionIconY;
+        if (spaceAvail < reactHeight) {
+            reactOffset = (spaceAvail - 360);
+        }
+        this.setState({showReactEmojiPicker: !this.state.showReactEmojiPicker, reactPickerOffset: reactOffset});
+    }
+
+    reactEmojiClick(emoji) {
+        this.setState({showReactEmojiPicker: false});
+        const emojiName = emoji.name || emoji.aliases[0];
+        addReaction(this.props.post.channel_id, this.props.post.id, emojiName);
+    }
+
     render() {
         const post = this.props.post;
         const flagIcon = Constants.FLAG_ICON_SVG;
         const mattermostLogo = Constants.MATTERMOST_ICON_SVG;
         const isSystemMessage = PostUtils.isSystemMessage(post);
 
+        if (post.state !== Constants.POST_FAILED &&
+            post.state !== Constants.POST_LOADING &&
+            !Utils.isPostEphemeral(post) &&
+            Utils.isFeatureEnabled(Constants.PRE_RELEASE_FEATURES.EMOJI_PICKER_PREVIEW)) {
+            this.canReact = true;
```

Comment:
The render() method should not be setting it, only rendering what's in state or props. It would be better to have this in a componentWillReceiveProps() function and then set canReact in state by doing `this.setState({canReact: true});` if your conditions are met